reyna/src/com/b2msolutions/reyna/services/ForwardService.java

```diff
@@ -37,21 +57,25 @@ protected void onHandleIntent(Intent intent) {
                 return;
             }
 
-            Message message = this.repository.getNext();
+            if (!this.messageProvider.canSend()) {
+                Logger.v(TAG, "ForwardService: messageProvider cannot send");
+                return;
+            }
+
+            Message message = this.messageProvider.getNext();
             while(message != null) {
 
                 this.thread.sleep(SLEEP_MILLISECONDS);
 
                 Logger.v(TAG, "ForwardService: processing message " + message.getId());
```

Comment:
Small thing, if i'm understanding this correctly, the ID logged here will be the first ID in the batch? Could this cause confusion if it skips logging for the rest of the batch? For example, you have batches 1 to 10, 11 to 20 and 21 to 30, the log will appear:
```"ForwardService: processing message 1"```
```*something happens and 11 to 20 isn't processed*```
```"ForwardService: processing message 21"```

This would just appear as 1 to 20 as being in one batch.

Could we log the first and last ID in a batch?